go/6_auth-kubernetes.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	vault "github.com/hashicorp/vault/api"
+)
+
+// Fetches a key-value secret (kv-v2) after authenticating to Vault with a Kubernetes service account.
+//
+// As the client, all we need to do is pass along the JWT token representing our application's Kubernetes Service Account in our login request to Vault.
+// This token is automatically mounted to your application's container by Kubernetes. Read more at https://www.vaultproject.io/docs/auth/kubernetes
+//
+// SETUP NOTES: If an operator has not already set up Kubernetes auth in Vault for you, then you must also first configure the Vault server with its own Service Account token to be able to communicate with the Kubernetes API
+// so it can verify that the client's service-account token is valid. The service account that will be performing that verification needs the ClusterRole system:auth-delegator.
+//
+//    export TOKEN_REVIEW_JWT=$(kubectl get secret $TOKEN_REVIEWER_SECRET --output='go-template={{ .data.token }}' | base64 --decode)
+//    export KUBE_HOST=$(kubectl config view --raw --minify --flatten --output='jsonpath={.clusters[].cluster.server}')
+//    kubectl config view --raw --minify --flatten --output='jsonpath={.clusters[].cluster.certificate-authority-data}' | base64 --decode > path/to/kube_ca_cert
+//
+//    vault write auth/kubernetes/config \
+//  	token_reviewer_jwt=${TOKEN_REVIEW_JWT} \
+//      kubernetes_host=${KUBE_HOST} \
+//      kubernetes_ca_cert=@path/to/kube_ca_cert \
+//      issuer="kubernetes/serviceaccount"
+//
+// The "issuer" field only needs to be set when running Kubernetes version 1.21 or above:
```

Comment:
This is normally correct, but not always. It's probably too complex to address fully within the scope you want to address, so I would just suggest updating the assertion a little bit, e.g. `The "issuer" field is normally only required when running Kubernetes 1.21 or above:`

For context, the fuller explanation is that the issuer will be something different from the default when using ephemeral tokens. This could be when pods are using projected service account token volumes, or it could be when using the `TokenCreate` endpoint like the Vault CSI provider does - however, it would be unusual for other generic app on k8s to use `TokenCreate` because the more common pattern is to use projected service account token volumes so that k8s manages the renewals. For projected token volumes, they became the default in 1.21.